### PR TITLE
Clear body tests

### PR DESCRIPTION
**Pull request by Pawkkie**
## Description
Wrote missing Clear Body tests.

This includes:

```
TO_DO_BATTLE_TEST("Clear Body prevents stat stage reduction from moves"); // Growl, Leer, Confide, Fake Tears, Scary Face, Sweet Scent, Sand Attack (Attack, Defense, Sp. Attack, Sp. Defense, Speed, Evasion, Accuracy
TO_DO_BATTLE_TEST("Clear Body prevents Sticky Web");
TO_DO_BATTLE_TEST("Clear Body doesn't prevent stat stage reduction from moves used by the user"); // e.g. Superpower
TO_DO_BATTLE_TEST("Clear Body doesn't prevent Speed reduction from Iron Ball");
TO_DO_BATTLE_TEST("Clear Body doesn't prevent Speed reduction from paralysis");
TO_DO_BATTLE_TEST("Clear Body doesn't prevent Attack reduction from burn");
TO_DO_BATTLE_TEST("Clear Body doesn't prevent receiving negative stat changes from Baton Pass");
TO_DO_BATTLE_TEST("Clear Body doesn't prevent Topsy-Turvy");
TO_DO_BATTLE_TEST("Clear Body doesn't prevent Spectral Thief from resetting positive stat changes");
TO_DO_BATTLE_TEST("Clear Body is ignored by Mold Breaker");
```

## **Discord contact info**
@Pawkkie

### CODE REVIEW

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:67]: Celebrate is implicit when not specified, so it can be removed:)

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:53]: I would expand this to multiple lines to avoid people from thinking they should stack multiple NONE_OF. Same for others.

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:79]: There's a function called HasMoveEffect or something like that should be used in case, as move effects can be in other positions besides the first one.

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:85]: Multiline

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:113]: Multiline

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:89]:
```suggestion
SINGLE_BATTLE_TEST("Mold Breaker ignores Clear Body")
```

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:92]: Let's add Teravolt and Turboblaze here.

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:130]: In tests, braces are in the same line as the if/else

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:127]: Implicit Celebrate

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:152]: Indentation

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:191]: Indentation

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:210]: Indentation

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:207]: Implicit Celebrate

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:229]: Indentation

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:222]: Implicit moves

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:202]: Implicit moves

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:184]: Implicit moves

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:145]: Implicit move

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:108]: Implicit move

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:81]: Implicit move

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:48]: Implicit move

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

File: test/battle/ability/clear_body.c
Comment (Pawkkie) [reply test/battle/ability/clear_body.c:248]: This one specifically, I can't remove the Moves(); sections without the compiler telling me the moves are invalid, in spite of the rest of the cleanup. Not sure why, let me know if I'm missing something obvious when you get to re-review :)

**Comment by Pawkkie** [comment]
I am once again unsure of why this is failing, but I've also discovered that both Full Metal Body and White Smoke are missing their tests, so even once this is fully sorted out don't merge it yet, just let me know that it's good to go and I'll copy paste all the tests over and make the necessary changes to cover both of those abilities as well :)

**Comment by Pawkkie** [comment]
Ready for re-review!

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

File: test/battle/ability/clear_body.c
Comment (Pawkkie) [reply test/battle/ability/clear_body.c:102]: this is so cool lol

**Comment by Pawkkie** [comment]
Feedback incorporated! Let me know if it's good before merging it so I can paste the tests over to Full Metal Body and White Smoke as well :)

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [reply test/battle/ability/clear_body.c:248]: @mrgriffin do you know why this might be?

Comment (AlexOn1ine) [reply test/battle/ability/clear_body.c:248]: You can't remove them from both mons or does it only give an error you remove it from one mon?

Comment (Pawkkie) [reply test/battle/ability/clear_body.c:248]: Removing from both mons or an individual mon all caused errors

Comment (AlexOn1ine) [reply test/battle/ability/clear_body.c:248]: weird

**Review by AlexOn1ine — CHANGES_REQUESTED** [review]
Most tests are missing an ability check. Whether if it goes of or not.  

Regarding the test that needs the moves specified. I would leave a comment for now. We shouldn't stall the PR because of it.

**Comment by Pawkkie** [comment]
> Most tests are missing an ability check. Whether if it goes of or not.
> 
> Regarding the test that needs the moves specified. I would leave a comment for now. We shouldn't stall the PR because of it.
> 
> /Edit Duke on discord discovered that the test passes fine with Metang and no moves. It could be a problem with Beldum being illiterate

What d'you mean by ability check, like d'you want me to parametrize the ability?

Would you prefer I leave that specific test as is with Beldum and a comment, or change Beldum for Metang? Or change to Metang throughout?

**Comment by AlexOn1ine** [comment]
> Would you prefer I leave that specific test as is with Beldum and a comment, or change Beldum for Metang? Or change to Metang throughout?

I think it is fine if you just replace Beldum with Metang without a comment

> What d'you mean by ability check, like d'you want me to parametrize the ability?

No, an ability check like like this inside `SCENE`
ability goes off
```
ABILITY_POPUP(opponent, ABILITY_INTIMIDATE);
```
and does not
```
NOT ABILITY_POPUP(opponent, ABILITY_INTIMIDATE);
```

**Comment by Pawkkie** [comment]
Ready for re-review, again let me know if this is good before merging so I can copy the tests over to the other two abilities :)

**Comment by AlexOn1ine** [comment]
> Ready for re-review, again let me know if this is good before merging so I can copy the tests over to the other two abilities :)

Looks good, Thx! 

@AsparagusEduardo could we get a re-review?

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

File: test/battle/ability/clear_body.c
Comment (Pawkkie) [reply test/battle/ability/clear_body.c:48]: Sweet Scent, this is how it was handled elsewhere in the tests :)

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:49]:
```suggestion
        ASSUME(gMovesInfo[MOVE_SWEET_SCENT].effect == (B_UPDATED_MOVE_DATA >= GEN_6 ? EFFECT_EVASION_DOWN_2 : EFFECT_EVASION_DOWN));
```

File: test/battle/ability/clear_body.c
Comment (AsparagusEduardo) [inline test/battle/ability/clear_body.c:134]:
```suggestion
        ASSUME(gMovesInfo[MOVE_SWEET_SCENT].effect == (B_UPDATED_MOVE_DATA >= GEN_6 ? EFFECT_EVASION_DOWN_2 : EFFECT_EVASION_DOWN));
```

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

**Review by AsparagusEduardo — DISMISSED** [review]

**Comment by AsparagusEduardo** [comment]
@AlexOn1ine just missing yours

File: test/battle/ability/clear_body.c
Comment (AlexOn1ine) [inline test/battle/ability/clear_body.c:221]:
```suggestion
        MESSAGE("Wobbuffet used Scary Face!");
        NOT ABILITY_POPUP(opponent, ABILITY_CLEAR_BODY);        
```

File: test/battle/ability/clear_body.c
Comment (AlexOn1ine) [inline test/battle/ability/clear_body.c:98]: I've watched the recording of the tests and it uses Armaldo and Kingler in place of Beldum. Could anybody confirm?

File: test/battle/ability/clear_body.c
Comment (AlexOn1ine) [inline test/battle/ability/clear_body.c:241]: The message doesn't prove if a move was used or not. It needs to check the animation as well. Same principle with the ability check 
move anim playes:
`ANIMATION(ANIM_TYPE_MOVE, MOVE_EMBER, player);`
does not 
`ANIMATION(ANIM_TYPE_MOVE, MOVE_EMBER, player);`

Also the not ability check should be in a places where it technically could go off.

**Review by AlexOn1ine — COMMENTED** [review]
Sorry Pawkkie, I found a couple more things 😅

I only made comments for two checks but it applies to some of the tests

File: test/battle/ability/clear_body.c
Comment (Pawkkie) [reply test/battle/ability/clear_body.c:98]: I haven't been able to reproduce this, my test.elf is still showing Beldum. Is it just for this specific test or for all of them?

Comment (Pawkkie) [reply test/battle/ability/clear_body.c:98]: Okay cool, no worries! Glad it got figured out :)

File: test/battle/ability/clear_body.c
Comment (Pawkkie) [reply test/battle/ability/clear_body.c:245]: Pushed this addition, let me know if I've missed places you'd have liked it added :)

**Review by AlexOn1ine — APPROVED** [review]
Perfect, thx. Sorry that you had to do so many revisions.

**Pull request merged**
